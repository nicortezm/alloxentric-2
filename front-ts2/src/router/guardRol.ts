import Swal from "sweetalert2"




export const Role1Guard = ( to , from , next ) => {
    
    return new Promise ( () => {
     
        const rol = localStorage.getItem('rol')
  
        if( rol === 'role1' ) {

            console.log( 'Esta autorizado a filtrar')
            next()

        } else  {

            console.log( 'Bloqueado por el GuardRol' )

            next({ name: 'home'})

            Swal.fire({
                icon: 'warning',
                title: 'Rol invalido',
                text: `${rol} no tiene acceso a esa vista`,
              })
        }
    })
}



export const Role2Guard = ( to , from , next ) => {
    
    return new Promise ( () => {
     
        const rol = localStorage.getItem('rol')
  
        if( rol === 'role2' ) {

            console.log( 'Esta autorizado a etiquetar')
            next()

        } else  {

            console.log( 'Bloqueado por el GuardRol' )
            
            Swal.fire({
                icon: 'warning',
                title: 'Rol invalido',
                text: `${rol} no tiene acceso a esa vista`,
              })

            next({ name: 'home'})

        }
    })
}


