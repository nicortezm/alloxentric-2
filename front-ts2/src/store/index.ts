import { createStore } from 'vuex'


export default createStore({
  state: {
    user: {
      isAuthenticated: false,
      name: "",
      idToken: "",
      accessToken: "",
      rol: ""
    }
  },
  getters: {
  },


  mutations: {
    
    logout(state) {
      state.user.isAuthenticated = false;
      state.user.name = "";
      state.user.idToken = "";
      state.user.accessToken = "";
      state.user.rol = ""
    },

    login(state, payload) {
      state.user.isAuthenticated = true;
      state.user.idToken = payload.idToken;
      state.user.accessToken = payload.accessToken;
    },

    setName(state, payload) {
      state.user.name = payload.name;
    }
  },



  actions: {
  },



  modules: {
  }
})
