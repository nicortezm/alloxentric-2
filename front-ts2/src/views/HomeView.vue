<template> 

  <div class="row p-1">
    <div class="col d-flex justify-content-end">
        <div class="px-2" v-if="isAuthenticated == true">
            <a class="text-dark mr-3">Bienvenido: <span class="text-uppercase">{{ $store.state.user.name }}</span></a>
            <br>
            <button class="btn btn-secondary mt-2 text-uppercase" @click="onLogout"><i class="fa-solid fa-arrow-left"></i> Logout</button>
        </div>        
    </div>
  </div>
  <div class="container mt-2 p-2 d-flex justify-content-center animate__animated animate__fadeInDown">
    <img class="p-2" src="@/assets/logoA.png"/>
    <div class="mt-5">
      <h1>Servicios de Audio</h1>
    </div>
  </div>

  <div class="container d-flex justify-content-center">
      <div class="col-lg-6 col-sm-12 border border-dark p-5" style="border-radius: 20px; background-color:#f1faee;">
      <h3 class="text-center">¿Qué Desea Hacer?</h3>
      <button class="filter btn btn-secondary mt-3" @click="toFilter"  style="border-radius: 10px;"> Filtros de Calidad</button>
      <button class="tag btn btn-secondary mt-3 ml-2" @click="toTags" style="border-radius: 10px;"> Etiquetar Audios</button>
    </div>
  </div>
</template>




<script lang="ts">

import { defineComponent } from 'vue';

export default defineComponent({

  name: 'HomeView',

  computed:{
    isAuthenticated(){
      return this.$store.state.user.isAuthenticated;
    }
  },

  methods: {
    toTags() {
      this.$router.push({ name: 'tags'})
    },

     toFilter() {
      this.$router.push({ name: 'filter'})
    },

    onLogout(){
      let payloadRefreshedTokens = {
        idToken: '',
        accessToken: ''
      }
      this.$store.commit("login", payloadRefreshedTokens)
      this.$store.commit("logout")
    },
    getUsername(){
      return this.$store.state.user.name
    }
  }


})
</script>

