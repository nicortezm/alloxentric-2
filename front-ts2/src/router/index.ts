import { createRouter, createWebHashHistory, RouteRecordRaw } from 'vue-router'
import HomeView from '../views/HomeView.vue'
import { Role1Guard, Role2Guard } from './guardRol';
import Keycloak from 'keycloak-js';




export const routes: Array<RouteRecordRaw> = [
  {
    path: '/',
    name: 'home',
    component: HomeView,
    props: Keycloak
  },
  {
    path: '/about',
    name: 'about',
    component: () => import(/* webpackChunkName: "about" */ '../views/AboutView.vue')
  },
  {
    path: '/filter',
    name: 'filter',
    beforeEnter: [ Role1Guard ], 
    component: () => import(/* webpackChunkName: "Filters" */ '../views/FilterView.vue'),
  },
  {
    path: '/tags',
    name: 'tags',
    beforeEnter: [ Role2Guard ], 
    component: () => import(/* webpackChunkName: "Tags" */ '../views/TagsView.vue')
  },
  {
    path: '/details/:endpoint',
    name: 'details',
    component: () => import(/* webpackChunkName: "Tags" */ '../views/DetailsView.vue'),
    props: ( route ) => {
      console.log( route )
      return {
          endpoint: route.params.endpoint
      }
    } 
  },

  { 
    path: '/:pathMatch(.*)*', 
    component: HomeView
  }    
]

const router = createRouter({
  history: createWebHashHistory(),
  routes,
})










export default router
