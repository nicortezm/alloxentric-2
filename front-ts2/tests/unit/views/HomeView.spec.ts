import Swal  from 'sweetalert2'
import HomeView from '@/views/HomeView.vue'

import { routes } from "@/router";
import { storeRole1, storeRole2 } from "../mockStore/store";
import { flushPromises, shallowMount } from "@vue/test-utils";
import { createRouter, createWebHistory } from 'vue-router'


const router = createRouter({
    history: createWebHistory(),
    routes: routes,
})

jest.mock('sweetalert2', () => ({
    fire: jest.fn(),
    showLoading: jest.fn(),
    close: jest.fn()
}))

describe('Pruebas en la Vista de Home View', () => {

    const mockRouter = {
        push: jest.fn(),
    }


    

    test('Debe de renderizar correctamente HomeView y mostrar el nombre de usuario ', () => {

        const wrapper = shallowMount( HomeView, {
            global: {
                plugins: [ storeRole1 ],
                mocks: {
                    $router: router,           
                }
            },
            computed: {
                isAuthenticated() {
                return storeRole1.state.user.isAuthenticated
                }
            },
        })

        const UserName = wrapper.find('span').text() 
        expect( UserName ).toBe('jorge' )
        expect( wrapper.html() ).toMatchSnapshot()
    });



    test('Al clickear botÃ³n de filtro se dirigira a filterView', () => {

        const wrapper = shallowMount( HomeView, {
            global: {
                plugins: [ storeRole1 ],
                mocks: {
                    $router: mockRouter,           
                }
            },
            computed: {
                isAuthenticated() {
                return storeRole1.state.user.isAuthenticated
                }
            },
        })

        wrapper.find('.filter').trigger('click')
        expect( mockRouter.push ).toHaveBeenCalledWith({ name: 'filter' })
    });


    
    test('Al clickear boton de tag con Role2 el guard  NO se dispara', async() => {
        
        localStorage.setItem('rol', storeRole2.state.user.rol )
        
        router.push({ name: 'home' })

        await router.isReady()

        const wrapper = shallowMount( HomeView, {
            global: {
            plugins: [ storeRole2, router ],

            },
            computed: {
                isAuthenticated() {
                return storeRole2.state.user.isAuthenticated
                }
            },
        })
        
        wrapper.find('.tag').trigger('click')     
        await flushPromises()

        expect( Swal.fire ).not.toHaveBeenCalled()      
    });


    test('Al clickear boton de filtro con Role1 el guard NO se dispara', async() => {
        
        localStorage.setItem('rol', storeRole1.state.user.rol )
        
        router.push({ name: 'home' })

        await router.isReady()

        const wrapper = shallowMount( HomeView, {
            global: {
            plugins: [ storeRole1, router ],

            },
            computed: {
                isAuthenticated() {
                return storeRole1.state.user.isAuthenticated
                }
            },
        })
        
        wrapper.find('.filter').trigger('click')     
        await flushPromises()

        expect( Swal.fire ).not.toHaveBeenCalled()      
    });

})



        