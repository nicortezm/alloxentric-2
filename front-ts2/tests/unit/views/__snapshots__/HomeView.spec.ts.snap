// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en la Vista de Home View Debe de renderizar correctamente HomeView y mostrar el nombre de usuario  1`] = `
<div class="row p-1">
  <div class="col d-flex justify-content-end">
    <div class="px-2"><a class="text-dark mr-3">Bienvenido: <span class="text-uppercase">jorge</span></a><br><button class="btn btn-secondary mt-2 text-uppercase"><i class="fa-solid fa-arrow-left"></i> Logout</button></div>
  </div>
</div>
<div class="container mt-2 p-2 d-flex justify-content-center animate__animated animate__fadeInDown"><img class="p-2">
  <div class="mt-5">
    <h1>Servicios de Audio</h1>
  </div>
</div>
<div class="container d-flex justify-content-center">
  <div class="col-lg-6 col-sm-12 border border-dark p-5" style="border-radius: 20px; background-color: rgb(241, 250, 238);">
    <h3 class="text-center">¿Qué Desea Hacer?</h3><button class="filter btn btn-secondary mt-3" style="border-radius: 10px;"> Filtros de Calidad</button><button class="tag btn btn-secondary mt-3 ml-2" style="border-radius: 10px;"> Etiquetar Audios</button>
  </div>
</div>
`;
